pkg/dns/azure/client/auth.go

```diff
@@ -46,7 +46,7 @@ func getAuthorizerForResource(config Config) (autorest.Authorizer, error) {
 	}
 
 	scope := config.Environment.TokenAudience
-	if !strings.HasPrefix(scope, "/.default") {
+	if !strings.HasSuffix(scope, "/.default") {
```

Comment:
Can we create a unit test case at all -- something that would have caught this change up front?

Reply:
A bit of both; if we had a test (or had written a test) would we have discovered the need to use HasSuffix (over HasPrefix) because the test would have failed upfront.

Reply:
Done